src/decode.h

```diff
@@ -250,7 +263,35 @@ static inline uint32_t sign_extend_b(const uint32_t x)
     return (int32_t) ((int8_t) x);
 }
 
+static inline bool queue_is_full(struct block_t *block)
+{
+    assert(block);
+    return (block->front - block->rear) == block->capacity;
+}
+
+static inline bool queue_is_empty(struct block_t *block)
+{
+    assert(block);
+    return block->front == block->rear;
+}
+
+static inline struct rv_insn_t *get_rear_mem_block(struct block_t *block)
```

Comment:
Apply consistent naming scheme for these helper functions.

Reply:
I think it could be renamed as following
```diff
-  static inline struct rv_insn_t *get_rear_mem_block(struct block_t *block)
+ static inline struct rv_insn_t *queue_get_rear_block(block_t *block)

- static inline struct rv_insn_t *alloc_mem_block(block_t *block)
+ static inline struct rv_insn_t *queue_alloc_block(block_t *block)

- static inline void free_mem_block(block_t *block)
+ static inline void queue_free_block(block_t *block)
```

Reply:
It is a bit weird that there is an operation for "rear" but not "front" counterpart. Instead, you can hide the operation "get_rear" by explicitly mentioning its usage. We can say something like `queue_current_block`.

Reply:
Both `queue_alloc_block` and `alloc_mem_block` are misleading. We can say, `block_enqueue`. Similarly, we can say, `block_dequeue` for the opposite.